CMakeLists.txt

```diff
@@ -1461,6 +1464,8 @@ if(CONFIG_OUTPUT_DISASSEMBLE_ALL)
     )
 endif()
 
+# probably some equivalent command exists to read stats from a Mach-O binary
+if (NOT (${CMAKE_HOST_SYSTEM_NAME} STREQUAL "Darwin" AND CONFIG_ARCH_POSIX))
```

Comment:
We should avoid testing for Darwin and ARCH_POSIX everywhere.

We already have an infrastructure in place regarding compiler and linker flags, as well as bintools commands / flags / scripts.
See more here:
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/toolchain
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/compiler
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/bintools
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/linker

Of course that is not directly related to host, mostly the toolchain / compiler / linker, but we do have `host-gnu` which basically means Linux (I haven't tested if `host` works on Windows, but I don't think it does).
So it could hint that `host-gnu` should support a subidentifier with system name, Linux, Darwin, Windows.

I believe we should try to see how host specific knowledge can fit into that design.
(Of course this code is fine for exploration work)